### PR TITLE
blog: Add the draft of the GSoC'24 UEFI GOP post

### PR DESCRIPTION
**Pull request by procub3r**
The first blog post outlining my progress in the first three weeks and the next steps.

### CODE REVIEW

File: content/blog/2024-06-18-gsoc-uefi-gop.mdx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+The widely available and standardized [UEFI Graphics Output Protocol](https://uefi.org/specs/UEFI/2.10/12_Protocols_Console_Support.html#efi-graphics-output-protocol) (GOP) interface is an excellent candidate. It provides a raw framebuffer, which can be used to set individual pixels on the screen.
+
+This project aims to implement a console based on UEFI GOP, which is capable of parsing fonts and displaying text on the screen by setting the appropriate pixels in the framebuffer.
+
+## Progress
+- I have implemented a low level driver that locates the GOP protocol and gets the framebuffer (and some metadata like width and height).
+
+- I've created a [convenience script](https://github.com/procub3r/unikraft/blob/poc/run.sh) that helps build a UEFI image and run it in qemu.
+
+- I have also written a small [proof of concept](https://github.com/procub3r/unikraft/tree/poc) that demonstrates our ability to control individual pixels on the screen:
```
Comment (mogasergiu) [inline content/blog/2024-06-18-gsoc-uefi-gop.mdx:29]: I would directly link the draft PR here.

File: content/blog/2024-06-18-gsoc-uefi-gop.mdx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+<img width="100px" src="https://summerofcode.withgoogle.com/assets/media/gsoc-generic-badge.svg" align="right" />
+
+Unikraft's ability to print logs to the screen is quite limited, which makes it harder to debug it during development. While it is entirely possible to use the serial port console, it is not as easily accessible or available on usual consumer machines.
+
+The widely available and standardized [UEFI Graphics Output Protocol](https://uefi.org/specs/UEFI/2.10/12_Protocols_Console_Support.html#efi-graphics-output-protocol) (GOP) interface is an excellent candidate. It provides a raw framebuffer, which can be used to set individual pixels on the screen.
+
+This project aims to implement a console based on UEFI GOP, which is capable of parsing fonts and displaying text on the screen by setting the appropriate pixels in the framebuffer.
+
+## Progress
+- I have implemented a low level driver that locates the GOP protocol and gets the framebuffer (and some metadata like width and height).
```
Comment (mogasergiu) [inline content/blog/2024-06-18-gsoc-uefi-gop.mdx:25]: > driver

I would just call it "routine"

File: content/blog/2024-06-18-gsoc-uefi-gop.mdx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+- I have also written a small [proof of concept](https://github.com/procub3r/unikraft/tree/poc) that demonstrates our ability to control individual pixels on the screen:
+
+  <Image
+        src="/images/uefi-gop-proof-of-concept.png"
+        title="UEFI GOP Proof of Concept"
+        description="A gradient displayed on the screen (in qemu-system-x86_64)"
+  />
+
+## Next Steps
+- Display characters from a bitmap font.
```
Comment (mogasergiu) [inline content/blog/2024-06-18-gsoc-uefi-gop.mdx:38]: What font have you chosen to start with?

File: content/blog/2024-06-18-gsoc-uefi-gop.mdx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+---
+
+## Project Overview
+<img width="100px" src="https://summerofcode.withgoogle.com/assets/media/gsoc-generic-badge.svg" align="right" />
+
+Unikraft's ability to print logs to the screen is quite limited, which makes it harder to debug it during development. While it is entirely possible to use the serial port console, it is not as easily accessible or available on usual consumer machines.
+
+The widely available and standardized [UEFI Graphics Output Protocol](https://uefi.org/specs/UEFI/2.10/12_Protocols_Console_Support.html#efi-graphics-output-protocol) (GOP) interface is an excellent candidate. It provides a raw framebuffer, which can be used to set individual pixels on the screen.
+
+This project aims to implement a console based on UEFI GOP, which is capable of parsing fonts and displaying text on the screen by setting the appropriate pixels in the framebuffer.
```
Comment (mogasergiu) [inline content/blog/2024-06-18-gsoc-uefi-gop.mdx:22]: Can you also describe how this framebuffer is fetched, what it actually is, its state after ExitBootServices is called, etc.

**Review by mogasergiu — CHANGES_REQUESTED** [review]

**Comment by StefanJum** [comment]
@procub3r can you please address the comments so we can merge and publish today?

**Comment by procub3r** [comment]
Yep I shall adress them and push within a couple of hours

File: content/blog/2024-06-18-gsoc-uefi-gop.mdx (hunk 3 above)
Comment (procub3r) [reply content/blog/2024-06-18-gsoc-uefi-gop.mdx:38]: https://github.com/dhepper/font8x8 looks cool. I haven't set it in stone yet though.

Comment (mogasergiu) [reply content/blog/2024-06-18-gsoc-uefi-gop.mdx:38]: "License: Public Domain" perfect :+1: 
What I had in mind was the 16x8 VGA font e.g. <https://www.inp.nsk.su./~bolkhov/files/fonts/univga/>
But if this 8x8 is clear enough, I would like it more as well :smile: .

Sure, let's give this 8x8 a shot.

File: content/blog/2024-06-18-gsoc-uefi-gop.mdx (hunk 4 above)
Comment (procub3r) [reply content/blog/2024-06-18-gsoc-uefi-gop.mdx:22]: I've written more about the framebuffer. Couldn't fit info about ExitBootServices into the flow though. Should I work on providing more context here?

File: content/blog/2024-06-18-gsoc-uefi-gop.mdx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+Text can be rendered on the screen by lighting up particular pixels. The exact pixels to light up to display a character is known by implementing a font parser, which derives this information from provided font(s).
+
+## Progress
+- I have implemented a low level routine that locates the GOP protocol and gets the framebuffer (and some metadata like width and height).
+
+- I've created a [convenience script](https://github.com/procub3r/unikraft/blob/poc/run.sh) that helps build a UEFI image and run it in qemu.
+
+- I have also written a small [proof of concept](https://github.com/procub3r/unikraft/tree/poc) that demonstrates our ability to control individual pixels on the screen. Link to the draft PR [here](https://github.com/unikraft/unikraft/pull/1448).
```
Comment (mogasergiu) [inline content/blog/2024-06-18-gsoc-uefi-gop.mdx:33]: I meant replace `https://github.com/procub3r/unikraft/tree/poc` with `https://github.com/unikraft/unikraft/pull/1448`

File: content/blog/2024-06-18-gsoc-uefi-gop.mdx (hunk 4 above)
Comment (procub3r) [reply content/blog/2024-06-18-gsoc-uefi-gop.mdx]: Update: I've brought in some context and talked about the aforementioned points :D

File: content/blog/2024-06-18-gsoc-uefi-gop.mdx (hunk 5 above)
Comment (procub3r) [reply content/blog/2024-06-18-gsoc-uefi-gop.mdx:33]: Oh welp. On it!

Comment (procub3r) [reply content/blog/2024-06-18-gsoc-uefi-gop.mdx:33]: Done :D

**Review by mogasergiu — APPROVED** [review]
@procub3r Once you remove the ":D" :laughing:  I would say LGTM.

Reviewed-by: Sergiu Moga <sergiu@unikraft.io>

**Comment by procub3r** [comment]
Done :D

**Review by StefanJum — APPROVED** [review]

**Pull request merged**